packages/client/tests/functional/_utils/getTestSuiteInfo.ts

```diff
@@ -182,15 +185,22 @@ export function getTestSuiteSchema(suiteMeta: TestSuiteMeta, matrixOptions: Name
 
   // if there's no preview features, append them to the default generator block
   if (previewFeatureMatch === null && defaultGeneratorMatch !== null) {
-    schema = schema.replace(defaultGeneratorMatch[0], `${defaultGeneratorMatch[0]}\n${previewFeaturesStr}`)
+    const replacement = `${defaultGeneratorMatch[0]}\n${previewFeaturesStr}`
+    schema = schema.replace(defaultGeneratorMatch[0], replacement)
+  }
+
+  // if an engine type is specified, append it to the default generator block
+  if (matrixOptions.engineType !== undefined && defaultGeneratorMatch !== null) {
+    const replacement = `${defaultGeneratorMatch[0]}\nengineType = "${engineType}"`
+    schema = schema.replace(defaultGeneratorMatch[0], replacement)
```

Comment:
This part is optional, I decided to properly write `engineType` into the schema. Optional in the sense that I could have used the env var instead. I prefer if it is as close as possible to reality.

Reply:
I am fine with either, but can we do it in the same way for binary and wasm engines too? Right now binary uses env var.

Reply:
Definitely. Part of my wider goal doing this also was that generally I consider the `engineType` to be one dimension of the  testing matrix. For a long time it has been an env var managed by the ci, that feels weird as other matrix elements are managed by the setup. I have put more structure in this matrix, making it clear and accessible on the type level and at runtime in tests, and making `getTestSuitePlan` the source of truth.

In other words, with the related changes of the PR, instead of doing
```ts
if (getClientEngineType() !== ClientEngineType.Binary) {
```
then you could have
```ts
if (suiteConfig.engineType === ClientEngineType.Library) {
```
(`suiteConfig` now has `provider`, `providerFlavor`, `relationMode`, and `engineType`)